File: code/java/canJump.java
class Solution {
    public boolean canJump(int[] nums) {
        int i = -1;
        int reach = 0;
        for(;i<nums.length && i < reach;){
            i++;
            if(reach >= nums.length - 1)
                return true;
            reach = Math.max(i+nums[i],reach);
        }
        return false;
    }
}

File: code/python/canJump.py
class Solution:
    def canJump(self, nums: List[int]) -> bool:
        i = -1
        reach = 0
        n = len(nums)
        while(i<n and i<reach): 
            i = i + 1
            if reach >= n-1:
                return True
            reach = max(i+nums[i],reach)
            

        return False
